test/test_mahalanobis_mixin.py

```diff
@@ -284,5 +284,14 @@ def test_transformer_is_2D(estimator, build_dataset):
 
     # test that it works for 1 feature
     trunc_data = input_data[..., :1]
+    # we drop duplicates that might have been formed, i.e. of the form
```

Comment:
I did this to make tests pass, they didn't because of #177 and #176 
The dataset created here is reasonable: we just drop duplicates which what a good user would do in practice. It's not like if we created a very weird dataset to prevent the tests from failing, so it's ok to do that. #177 and #176 need to be fixed and tested separately in other PRs